### PR TITLE
PROD-30875: Implement "event published" event for the EDA

### PR DESCRIPTION
**Pull request by tregismoreira**
## Description

This pull request introduces the new EDA event "event publish" which is dispatched to the Kafka topic `com.getopensocial.cms.event.publish`. The event is triggered on `hook_social_core_event_published`, a new custom hook also provided by this PR;

Here's an example of the generated payload:

```
{
	"specversion": "1.0",
	"id": "62d4fc19-88b1-4e33-a82c-47657696388b",
	"source": "/admin/config/system/cron/jobs/scheduler_cron/run",
	"type": "com.getopensocial.cms.event.publish",
	"datacontenttype": "application/json",
	"time": "2024-10-03T13:36:41Z",
	"data": {
		"event": {
			"id": "c42d8ff9-a2a4-4e79-8312-d3ecb8bb1af5",
			"created": "2024-10-03T13:36:41",
			"updated": "2024-10-03T15:36:17",
			"status": "published",
			"label": "xzczxc",
			"visibility": "community",
			"group": null,
			"author": {
				"id": "b0dc9a92-7e00-425d-a307-e6fb8418a69b",
				"displayName": "admin",
				"href": {
					"canonical": "http://cablecar.localhost/user/1/home"
				}
			},
			"allDay": true,
			"start": "2024-10-03T00:00:00",
			"end": "2024-10-18T00:00:00",
			"timezone": "UTC",
			"address": {
				"label": "",
				"countryCode": "",
				"administrativeArea": "",
				"locality": "",
				"dependentLocality": "",
				"postalCode": "",
				"sortingCode": "",
				"addressLine1": "",
				"addressLine2": ""
			},
			"enrollment": {
				"enabled": true,
				"method": "request"
			},
			"href": {
				"canonical": "http://cablecar.localhost/event/xzczxc"
			},
			"type": null
		},
		"actor": {
			"application": {
				"id": "123e4567-e89b-12d3-a456-426614174000",
				"name": "Cron Application"
			},
			"user": null
		}
	}
}
```

## Issue tracker
https://getopensocial.atlassian.net/browse/PROD-30875

## Theme issue tracker
N/A

## How to test
- [ ] Checkout branch `issue/PROD-30875-eda-event-published` of 'open_social'
- [ ] Enable the modules `social_eda` and `social_eda_dispatcher`
- [ ] Publish an Event by:
  - [ ] Manually changing the status in the node form
  - [ ] Using the bulk operations in `/admin/content` page
  - [ ] Scheduling the publication using `social_scheduler`

The message should be dispatched to Kafka. If you are running locally using the Docker containers, you should be able to see the message on the Kafka UI at http://localhost:8080.

## Release notes
N/A

## Change Record

Two new custom hooks were introduced:
- `hook_social_core_ENTITY_TYPE_published()`: fired when an entity of the given type is published
- `hook_social_core_ENTITY_TYPE_unpublished()`: fired when an entity of the given type is unpublished

The new hooks will be used in many EDA events we create.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

File: modules/social_features/social_event/src/EdaHandler.php [hunk 1]
```diff
@@ -109,4 +171,45 @@ public function fromEntity(NodeInterface $node): CloudEvent {
...
+
+  /**
+   * Dispatches the event.
+   *
+   * @param string $topic_name
+   *   The topic name.
+   * @param \CloudEvents\V1\CloudEvent $event
+   *   The CloudEvent event.
+   *
+   * @return void
```
Comment (ribel) [inline modules/social_features/social_event/src/EdaHandler.php:209]: @tregismoreira I see [PHPCS](https://github.com/goalgorilla/open_social/actions/runs/11167591676/job/31044319498?pr=4104) check failed on `Description for the @return value is missing`

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

File: modules/social_features/social_event/src/EdaHandler.php (hunk 1 above)
Comment (tregismoreira) [reply modules/social_features/social_event/src/EdaHandler.php]: @ribel fixed. Thanks ;)

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4104-cwlditqjnhzugkxmj68p5ys5eoqbpth9.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66fea0495f0b5f11b44c4f46

**Pull request merged**